communication/data_logging.hpp

```diff
@@ -280,15 +299,58 @@ class Data_logging
     State& state_;                              ///< The pointer to the state structure
 };
 
+
+/**
+ * \brief   Log data in files
+ *
+ * \tparam  N   Maximum number of variables to log
+ */
+template<uint32_t N>
+class Data_logging_tpl: public Data_logging
```

Comment:
It stands for template. @basil-huber I would like your opinion on the naming convention.

Here, most of the logic is implemented in the base class `Data_logging`. The child class `Data_logging_tpl` just holds the data. The template is the size of the array.

Pros: 
- we can keep most of the code in the *.cpp file and have nice error messages from the compiler (no template errors)
- other modules use the base class and do not need to be template

Cons:
- the name of the class you actually use is *_tpl